src/main.rs

```diff
@@ -34,25 +47,36 @@ fn cli() -> Command {
         )
 }
 
-fn main() {
+#[tokio::main]
+async fn main() -> Result<()> {
     let matches = cli().get_matches();
 
     match matches.subcommand() {
-        Some(("reconstruct", sub_matches)) => {
-            match sub_matches.subcommand() {
-                Some(("l1", args)) => {
-                    let start_block = args.get_one::<u64>("start-block").expect("required");
-                    let block_step = args.get_one::<u64>("block-step").expect("required");
-                    println!("reconstruct from L1, starting from block number {}, processing {} blocks at a time", start_block, block_step);
-                    // TODO(tuommaki): Implement block fetch logic.
-                }
-                Some(("file", args)) => {
-                    let input_file = args.get_one::<String>("FILE").expect("required");
-                    println!("reconstruct from file (path: \"{}\")", input_file);
-                }
-                _ => unreachable!(),
+        Some(("reconstruct", sub_matches)) => match sub_matches.subcommand() {
+            Some(("l1", args)) => {
+                let start_block = args.get_one::<u64>("start-block").expect("required");
+                let block_step = args.get_one::<u64>("block-step").expect("required");
+                let http_url = args.get_one::<String>("http-url").expect("required");
+                println!("reconstruct from L1, starting from block number {}, processing {} blocks at a time", start_block, block_step);
+
+                let fetcher = L1Fetcher::new(http_url)?;
+                let (tx, mut rx) = mpsc::channel::<Arc<Vec<CommitBlockInfoV1>>>(5);
+                tokio::spawn(async move {
+                    while let Some(blks) = rx.recv().await {
+                        blks.iter().for_each(|x| println!("{:?}", x));
+                    }
+                });
+
+                fetcher.fetch(tx, Some(U64([*start_block])), None).await?;
```

Comment:
I wrote the body here for now, to have ability to wire & test everything. I'd like to have this functionality somewhere else, but couldn't immediately come up with a good solution.

Should we put this to just separate function here or have a separate intermediate module that drives this logic?

Reply:
No need to refactor anything here yet, but one alternative could be to move the receiver block into some generic processor that will implement the actual storage (or printing as it is now)